### PR TITLE
Daily and Hourly Forecast Closes Issue #1, #10 and #24

### PR DESCRIPTION
**Pull request by saikeshari**
@grace-omotoso Please review

### CODE REVIEW

**Comment by grace-omotoso** [comment]
Please, fix conflicts

**Comment by saikeshari** [comment]
@grace-omotoso Can i look into the issue of loader mentioned by @sumana2001? I think I know how to fix it.
If it will be fixed, there would no conflicts in this pull

**Comment by grace-omotoso** [comment]
please, fix conflict

**Comment by grace-omotoso** [comment]
@saikeshari Please, fix conflicts

**Comment by saikeshari** [comment]
Fixed the conflicts @grace-omotoso

**Comment by grace-omotoso** [comment]
@saikeshari Please, fix conflicts

**Comment by grace-omotoso** [comment]
@saikeshari please, fix conflict

**Comment by grace-omotoso** [comment]
@Young-Einstein10, please, review this PR

**Comment by grace-omotoso** [comment]
@saikeshari daily forecast cards appear too scanty. Can we have more content e.g
weekly forecast
Also, can you add some effects to this cards.

On the main weather card, please, reduce the width and let us have current time displayed there

**Comment by GrayFlash** [comment]
Screenshot from 2021-11-19 16-06-10

Hey, I'm unable to see changes in cards and map after changing name of city. Please check it once. @saikeshari

**Review by Young-Einstein10 — COMMENTED** [review]
@saikeshari I noticed the weather details for a location are no longer updated whenever a new location is entered in the search bar. Could you look into it?

Also, I was thinking if this chart could have a light background just like the other cards, it would make it more visible to the eyes.

image

**Comment by grace-omotoso** [comment]
@saikeshari cards are not being updated when a new city is entered

**Comment by saikeshari** [comment]
Yes, I was working on that only.
Actually, these changes were to be made at root of project in App.js and useWeather, so encountered many problems, had to also introduce a new API to get langitude and latitude from a place.
Also due to this, I had to solve many merge conflicts.
But finally done ✨

**Comment by saikeshari** [comment]
In the deploy, it is still not working. On my local machine, it works.
Will update asap

**Comment by saikeshari** [comment]
@grace-omotoso Fixed it. Please review

File: src/components/WeatherCard.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    return (
+      <>
+            <div className="weather-card">
+              <div className="weatherCard-current ">
+                <h2>Weather in {city.name}, {city.sys.country}</h2>
+                <div className="weather-currentInner">
+                  <div className="weathercurrent-left">
+                  {weatherIcon(city.weather[0].icon)}
+                  {/* <img className="weatherCard-current-img" src={"https://openweathermap.org/img/wn/" + city.weather[0].icon + ".png"} /> */}
```
Comment (grace-omotoso) [inline src/components/WeatherCard.js]: remove commented code

File: src/helpers/DailyForecast.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    index === 0 || index === 7 ? <></> : 
+                    <div className="daily-div">
+                        <h3>
+                            {dayss[t.getDay()]}, {t.getDate()} {month[t.getMonth()]}
+                        </h3>
+                        
+                        <div className="dailyweather-icondiv">
+                            {weatherIcon(dayWeather.weather[0].icon)}
+                            <span>{avgTemp.toPrecision(4)} °C</span>
+                            {/* <img className="dail-forecast-img" src={"https://openweathermap.org/img/wn/" + dayWeather.weather[0].icon + ".png"} /> */}
```
Comment (grace-omotoso) [inline src/helpers/DailyForecast.js]: remove commented code

**Comment by grace-omotoso** [comment]
@saikeshari please, remove all commented code

**Comment by akshitadixit** [comment]
closes #1 #10 #24

**Comment by grace-omotoso** [comment]
@saikeshari Apologies, please, fix conflict again

**Comment by grace-omotoso** [comment]
@saikeshari Please, can we get the local time on the main weather info card. If it will be a quick one, please, include that. Otherwise, we can have a separate issue. Just let me know.

**Comment by saikeshari** [comment]
@grace-omotoso Added the time in the main card. Please review.

**Comment by saikeshari** [comment]
Right now, it is displaying the time at which the request was made. If continuously changing current time would be more suitable, let me know. Will do that

File: src/assets/css/weatherCard.css [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+.weatherCard-current{
+    width: 45%;
+    height: 100%;
+    display: flex;
+    margin: 1em;
+    flex-direction: column;
+    flex-wrap: wrap;
+    border-radius: 20px;
+    background: rgba(255, 255, 255, 0.1);
+    /* backdrop-filter: blur(35px); */
```
Comment (grace-omotoso) [inline src/assets/css/weatherCard.css]: Please, remove commented code.

File: src/helpers/customHooks/useWeather.js [hunk 4]
```diff
@@ -78,30 +117,22 @@ const useWeather = () => {
...
-              setIsLoaded(true);
-              setResults(result);
-
-              // Scroll To Top if request is successful
-              window.scrollTo(0, 0);
-            }
+            setIsLoaded(true);
+            setResults(result);
+            setIsLoading(false);
+            // setError(null);
```
Comment (grace-omotoso) [inline src/helpers/customHooks/useWeather.js]: remove commented code please

**Review by grace-omotoso — COMMENTED** [review]
Please, remove commented code

**Comment by saikeshari** [comment]
Removed the commented code

**Review by grace-omotoso — COMMENTED** [review]
Will merge but the time on the main card needs rework. Only shows 16:50 regardless of the selected location

**Pull request merged**